### ISSUE TITLE
Usage in public repo

### ISSUE THREAD

**Issue by iamolegga** [ArtiomTr/jest-coverage-report-action#194]
### Describe

Could not find any info on how to use this properly for public repo's PRs. To quote the page of settings/secret:

> Secrets are not passed to workflows that are triggered by a pull request from a fork. [Learn more](https://docs.github.com/en/actions/security-guides/encrypted-secrets).

You can find example error [here](https://github.com/iamolegga/create-nestjs-middleware-module/runs/4187588192?check_suite_focus=true)

Details:

<img width="865" alt="image" src="https://user-images.githubusercontent.com/5501657/141438759-5dcac8a2-620c-479e-b379-1b660fbc1f8e.png">

[Github action settings](https://github.com/iamolegga/create-nestjs-middleware-module/blob/master/.github/workflows/pr-coverage.yaml)

This is a quite useful action and I that would be really handy to use it in public repos.

Thanks in advance

**Comment by ArtiomTr** [comment]
Hello @iamolegga :wave:,

Unfortunately, this action cannot work properly in public repositories. At the moment, I see only one solution to this problem - an almost complete rewrite of the action and hosting it on private servers. Why? It's a bit complicated question, but I'll try to explain it.

To create comment under PR, this action requires an access token with `repo` permission. In public repository, each person could create a PR, and change `.yml` file in it. If action with modified code would run, this means it will have an access token, and possibly could do anything with it. This means people could write malicious code and take control of your repository. So GitHub doesn't let you “shoot yourself in the foot” by simply not passing any secrets to the actions performed from the fork.

How to solve this problem? GitHub provides perfect feature - [apps](https://docs.github.com/en/developers/apps/getting-started-with-apps/about-apps). If simplifying, this allows to set up an **custom** server, which will interact with GitHub. As a result, all the logic will be moved to the server, and when the PR triggers an action, the request will be sent to **custom** server, where it will be processed. There is multiple issues with that approach:
1. Code can no longer be open source, because code should integrate with private API like authentication, payments, etc.
2. Since the servers are not free, I will have to figure out how to make money from this action. However, this is not so easy, because simply applying a subscription model or any other does not make any sense for this action, since it is absolutely non-competitive compared to other large systems ([Codecov](https://about.codecov.io/), [Coveralls](https://coveralls.io/)). 

So to sum up, for now, I don't see any workaround for your issue. The solution of re-writing this action as GitHub app is not interesting for me, because the purpose of this repository is to be completely open-source and transparent, that anyone can fork / change / update / it.

I'll close this issue for now. If you have any ideas, feel free to express them, I will be happy to discuss.

**Comment by iamolegga** [comment]
Thank you for your clarification! I really appreciate that!

I believe I'm not the first here who faced this limitation, maybe it would be helpful for the others to add this (in a few words) to the docs?

Thanks anyway!

**Comment by ArtiomTr** [comment]
@iamolegga cool idea! I'll describe the situation about usage in public repositories in the `README.md` file

**Comment by HarelM** [comment]
I just found out about this limitation, our library is public and the main flow of code inwards is from PRs from forks.
Having said that, we have a flow that reports bundle size on each PR that can be seen here:
https://github.com/maplibre/maplibre-gl-js/blob/668a97d303351aadb80fca34f2417a08e49d7c1f/.github/workflows/analyze.yml
It works on forks as well. Is there a way to leverage part of the code there somehow to solve this problem?

**Comment by ArtiomTr** [comment]
Hello @HarelM :wave:,

I can't remember why GitHub downgrades token's privileges, but I think that's because this action uses a "composite" runner. https://github.com/ArtiomTr/jest-coverage-report-action/blob/253a5ba29b4b0e3834a94df8efdc2f13e236aba9/action.yml#L50-L54
Another option is to use a "node" runner. However, that would lock the node version and make this action useless. 

However, after looking at your workflow file, I have an idea: the action could generate only report *text*, and then another action (like `marocchino/sticky-pull-request-comment` in your workflow) could attach report text to the PR. That's not the best solution, but something.

I'm reopening this issue. Any ideas on how to improve action behaviour on public forks are very appreciated.

**Comment by HarelM** [comment]
I had the same idea after looking into this flow so I though it might be a valid workaround to share here.

**Comment by ArtiomTr** [comment]
@HarelM, I've released version, so the new `output` option is now available. I also updated the documentation, and [usage section](https://github.com/ArtiomTr/jest-coverage-report-action#usage) now contains an example with `sticky-pull-request-comment` action.

Could you please verify, if the action now works? I can open a PR in your repo, to test how it works with PRs from forks without "write" permission.

**Comment by HarelM** [comment]
Thanks for the update! I'll take a look into it.
